pkg/volume/util/util.go

```diff
@@ -96,29 +97,43 @@ func UnmountPath(mountPath string, mounter mount.Interface) error {
 // IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
 // IsNotMountPoint is more expensive but properly handles bind mounts.
 func UnmountMountPoint(mountPath string, mounter mount.Interface, extensiveMountPointCheck bool) error {
-	if pathExists, pathErr := PathExists(mountPath); pathErr != nil {
+	pathExists, pathErr := PathExists(mountPath)
+	return unmountMountPath(mountPath, mounter, extensiveMountPointCheck, pathExists, pathErr)
```

Comment:
Feel free to ignore this comment since I have not worked in this pkg before. However, it seems like an unusual pattern to check if the path exists here and, rather than verifying the result here, instead pass the result of `PathExists` check to `unmountMountPath`. What is the advantage of splitting the logic this way? What is the disadvantage if `unmountMountPath` calls `PathExists`?

Reply:
+1

Reply:
just for easy to write a unit test
we can pass a mock error to the parameter `pathErr`

Reply:
+1 to @jeffvance, if the path does not exist, it does not make sense to call `unmountMountPath` at all.